drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -172,6 +172,70 @@ def check_mutually_exclusive_attributes(document)
     link_check.call(document)
   end
 
+  def check_condition(condition, item_hash, link_id)
+    return unless condition.key?('question') # Not currently validating conditions with local constants; see below
+
+    # Find the referenced question. It should exist, but don't raise if not; that's checked elsewhere
+    referenced_question = item_hash[condition['question']]
+    return if referenced_question.nil?
+
+    return unless referenced_question['pick_list_options'] || referenced_question['pick_list_reference']
+
+    if referenced_question['pick_list_reference']
+      begin
+        options = Types::Forms::PickListOption.
+          options_for_type(referenced_question['pick_list_reference'], user: nil).
+          map(&:deep_stringify_keys)
+      rescue ArgumentError
+        # TODO - discuss, likely not worth it?
```

Comment:
Let's discuss the picklist_reference strategy. As mentioned in the ticket, we may decide to descope it for now.

This seems to work for picklists that are returned by the Graphql query `GetPickListType`, although I have some concerns that we're using that in a way it wasn't intended, and maybe some picklists break since we are not passing in a `user`.

Meanwhile, we would need to come up with an alternate solution for the case (currently ignored) where the picklist is based on an enum that the frontend resolves, never using `GetPickListQuery`.

Reply:
Thanks for the hints Gig! This is interesting, it partially works, but there is some magic going on with a few specific enum types `ProjectType` and `ProjectCompletionStatus` that I haven't been able to crack. I can keep working on it a bit tomorrow. Time spent so far about 5h (which was the original estimate), so I'll timebox to another hour tomorrow.

Reply:
@gigxz I think this may be okay to merge, but I'll hold off until you are back. Feel free to merge while I am out, or push more changes on this branch before merging.